plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1312 @@
+	serverName string,
+) error {
+	adr := OrderedMap{}
+
+	if execAsDomain {
+		adr = OrderedMap{
+			{"host", host},
+			{"server", serverName},
+			{"subsystem", "transactions"},
+		}
+	} else {
+		adr = OrderedMap{
+			{"subsystem", "transactions"},
+		}
+	}
+
+	bodyContent, err := h.createRequestBody(GetTransactionStat, adr)
+	if err != nil {
+		return fmt.Errorf("error on request to %s : %s\n", serverURL, err)
+	}
+
+	out, err := h.doRequest(serverURL, bodyContent)
+
+	if err != nil {
+		return fmt.Errorf("error on request to %s : %s\n", serverURL, err)
+	} else {
+		server := TransactionResponse{}
+		if err = json.Unmarshal(out, &server); err != nil {
+			return fmt.Errorf("Error decoding JSON response: %s : %s", out, err)
+		}
+
+		fields := make(map[string]interface{})
+		for key, value := range server.Result {
+			if strings.Contains(key, "number-of") {
+				fields[key], _ = strconv.ParseInt(value.(string), 10, 64)
+			}
+		}
+		tags := map[string]string{
+			"jboss_host":   host,
+			"jboss_server": serverName,
+		}
+
+		acc.AddFields("jboss_transaction", fields, tags)
+	}
+
+	return nil
+}
+
+// Gathers web data from a particular host
+// Parameters:
+//     acc      : The telegraf Accumulator to use
+//     serverURL: endpoint to send request to
+//     execAsDomain: JBoss runs in domain
+//     host     : the host being queried
+//     server   : the server being queried
+//
+// Returns:
+//     error: Any error that may have occurred
+
+func (h *JBoss) getWebStatistics(
+	acc telegraf.Accumulator,
+	serverURL string,
+	execAsDomain bool,
+	host string,
+	serverName string,
+	connector string,
+) error {
+	adr := OrderedMap{}
+	if execAsDomain {
+		adr = OrderedMap{
+			{"host", host},
+			{"server", serverName},
+			{"subsystem", "web"},
+			{"connector", connector},
+		}
+	} else {
+		adr = OrderedMap{
+			{"subsystem", "web"},
+			{"connector", connector},
+		}
+	}
+
+	bodyContent, err := h.createRequestBody(GetWebStat, adr)
+	if err != nil {
+		return fmt.Errorf("error on request to %s : %s\n", serverURL, err)
+	}
+
+	out, err := h.doRequest(serverURL, bodyContent)
+
+	if err != nil {
+		return fmt.Errorf("error on request to %s : %s\n", serverURL, err)
+	}
+	server := WebResponse{}
+	if err = json.Unmarshal(out, &server); err != nil {
+		return fmt.Errorf("Error decoding JSON response (WebResponse): %s : %s", out, err)
+	}
+
+	fields := make(map[string]interface{})
+	for key, value := range server.Result {
+		switch key {
+		case "bytesReceived", "bytesSent", "requestCount", "errorCount", "maxTime", "processingTime":
+			if value != nil {
+				switch value.(type) {
+				case int:
+					fields[key] = value.(float64)
+				case float64:
+					fields[key] = value.(float64)
+				case string:
+					f, err := strconv.ParseFloat(value.(string), 64)
+					if err != nil {
+						log.Printf("E! JBoss Error decoding Float  from string : %s = %s\n", key, value.(string))
+					} else {
+						fields[key] = f
+					}
+				}
+			}
+		}
```

Comment:
This is hard to read. How about
```suggestion
		if value == nil {
			continue
		}

		switch key {
		case "bytesReceived", "bytesSent", "requestCount", "errorCount", "maxTime", "processingTime":
			switch v := value.(type) {
			case int:
				fields[key] = float64(v)
			case float64:
				fields[key] = v
			case string:
				if f, err := strconv.ParseFloat(v, 64); err == nil {
					fields[key] = f
				}
			}
		}
```